src/main/java/com/baeldung/spring/SecSecurityConfig.java

```diff
@@ -150,7 +150,7 @@ public RememberMeServices rememberMeServices() {
 
     @Bean(name="GeoIPCountry")
     public DatabaseReader databaseReader() throws IOException, GeoIp2Exception {
-        final File resource = new File("src/main/resources/maxmind/GeoLite2-Country.mmdb");
+        final File resource = new File(this.getClass().getClassLoader().getResource("maxmind/GeoLite2-Country.mmdb").getFile());
```

Comment:
Hi @marioja 
1.Can we add null checks here for the 'resource' that will contribute in creating our bean?
2.Also, could you check if using ClassPathResource does also the job, since it seems more elegant approach?
3.Finally, could you build / run integration tests / check app runtime or standalone app and inside the web container, just to ensure nothing is broken?
Thanks